### PR TITLE
[WIP] some fixes?

### PR DESCRIPTION
**Pull request by sashei**
Connected to #144

### CODE REVIEW

File: app/src/main/java/mozilla/lockbox/store/DataStore.kt [hunk 1]
```diff
@@ -158,9 +158,17 @@ open class DataStore(
     private fun reset() {
         clearList()
         backend.reset().then({
+            if (backend is MemoryLoginsStorage) {
+                updateList()
+            }
+
```
Comment (jhugman) [inline app/src/main/java/mozilla/lockbox/store/DataStore.kt]: Yeah, this is kinda weird. I think we should probably be calling a `DataStoreAction.Sync` at some point.

File: app/src/main/res/layout/fragment_app_startup_placeholder.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+                                             android:layout_width="match_parent"
+                                             android:layout_height="match_parent"
+                                             android:id="@+id/fragment_app_startup_placeholder">
+
+    <TextView
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:text="I AM THE PLACEHOLDER"/>
```
Comment (jhugman) [inline app/src/main/res/layout/fragment_app_startup_placeholder.xml]: Twisted Placeholder.

Comment (sashei) [reply app/src/main/res/layout/fragment_app_startup_placeholder.xml]: 🎃

**Comment by sashei** [comment]
per conversation with @jhugman, he will take these changes and fixup the remaining issues which are unrelated to the UI tests.

**Comment by sashei** [comment]
changes incorporating, closing this.

**Pull request closed without merging**